reach

```diff
@@ -115,10 +117,81 @@ run_s () {
   sh -ac "$TMP/out.sh" "$0"
 }
 
+package_manager () {
+  if ( which apt > /dev/null 2>&1 ); then
+    export PACKAGE_MANAGER="apt"
+  fi
+
+  if ( which yum > /dev/null 2>&1 ); then
+    export PACKAGE_MANAGER="yum"
+  fi
+}
+
+install_make () {
+  package_manager
+
+  sudo ${PACKAGE_MANAGER} install make
+}
+
+install_curl () {
+  PACKAGE_MANAGER=package_manager
+
+  sudo $PACKAGE_MANAGER install curl 
+}
+
+install_docker () {
+  WINGET_VERSION="v1.3.431"
+  WINGET_ARTIFACT_NAME="Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle"
+  WINGET_BUNDLE_URL="https://github.com/microsoft/winget-cli/releases/download/${WINGET_VERSION}/${WINGET_ARTIFACT_NAME}"
+  TEMP_DIR="/mnt/c/tmp"
+  WIN_TEMP_DIR="c:\\tmp"
+  DOCKER_BIN="/mnt/c/Program Files/Docker/Docker/resources/bin/docker"
+  PWD=$(pwd)
+
+  cd / #Depending on the directory the user is, windows has a hard time with the path.
+
+  if (which wsl.exe > /dev/null 2>&1); then
+    printf "Installing Winget Windows Package Manager..."
+    if (! which winget.exe > /dev/null 2>&1); then
+      if [ ! -e $TEMP_DIR ]; then
+        mkdir $TEMP_DIR
+      fi
+      sudo wget $WINGET_BUNDLE_URL -O "${TEMP_DIR}/${WINGET_ARTIFACT_NAME}"; powershell.exe "cd ${WIN_TEMP_DIR}; powershell .\\${WINGET_ARTIFACT_NAME}"
+      while ( ! which winget.exe > /dev/null 2>&1 ); do
+        sleep 1;
+      done
```

Comment:
This infinite loop feels dangerous

Reply:
You're right.

I added a breaker to the while loops.